src/main/java/com/vmware/vipclient/i18n/VIPCfg.java

```diff
@@ -68,44 +64,19 @@
     public static final String         CACHE_L3      = "CACHE_L3";
     public static final String         CACHE_L2      = "CACHE_L2";
 
-    public boolean isSubInstance() {
-        return isSubInstance;
-    }
-    
-    public void setSubInstance(boolean subInstance) {
-        isSubInstance = subInstance;
-    }
-
-    private boolean isSubInstance = false;
-
-    private VIPCfg() {
-
-    }
-
     /**
-     * create a default instance of VIPCfg
-     * 
-     * @return
+     * @deprecated Use the {@link VIPCfgFactory#getCfg() getCfg} method.
      */
-    public static synchronized VIPCfg getInstance() {
-        if (gcInstance == null) {
-            gcInstance = new VIPCfg();
-        }
-        return gcInstance;
+    public static VIPCfg getInstance() {
+        return VIPCfgFactory.getCfg();
     }
 
     /**
-     * create a default instance of VIPCfg
-     *
-     * @return
+     * @deprecated Use the {@link VIPCfgFactory#getCfg(String) getCfg} method.
      */
     public static synchronized VIPCfg getSubInstance(String productName) {
```

Comment:
synchronized is redundant?

Reply:
This is deprecated anyway. Do we still want to change this?

Reply:
It's better to change because the lock has been moved to other places.

Reply:
sure